Modules/_sha3/sha3module.c

```diff
@@ -263,6 +273,8 @@ SHA3_dealloc(SHA3object *self)
         PyThread_free_lock(self->lock);
     }
     PyObject_Del(self);
+
+    Py_DECREF(Py_TYPE(self));
```

Comment:
use after free(`PyObject_Del(self)`), no?

Reply:
Oh, right! It may explain why test_hashlib fail with a Python built in debug mode.